apps/avatax/src/modules/avatax/calculate-taxes/avatax-calculate-taxes-payload-lines-transformer.ts

```diff
@@ -5,6 +5,71 @@ import { AvataxTaxCodeMatches } from "../tax-code/avatax-tax-code-match-reposito
 import { AvataxCalculateTaxesTaxCodeMatcher } from "./avatax-calculate-taxes-tax-code-matcher";
 import { avataxShippingLine } from "./avatax-shipping-line";
 
+type TaxBaseLine = TaxBaseFragment["lines"][0];
+
+const getUndiscountedTotalPrice = (line: TaxBaseLine) => {
+  if (line.sourceLine.__typename === "CheckoutLine") {
+    return line.sourceLine.undiscountedTotalPrice.amount;
+  }
+
+  return line.sourceLine.undiscountedTotalPrice.net.amount;
+};
+
+const checkIfIsDiscountedLine = (isDiscounted: boolean, line: TaxBaseLine) => {
+  if (isDiscounted) {
+    const undiscountedTotalPrice = getUndiscountedTotalPrice(line);
+    const totalPrice = line.totalPrice.amount;
+
+    return totalPrice !== undiscountedTotalPrice;
+  }
+
+  return false;
+};
+
+/**
+ * If entire checkout discount then lines in webhook subscription are not discounted
+ * If one item is discounted then the line has discounted totalPrice
+ */
+const checkDiscounts = (taxBase: TaxBaseFragment, isDiscounted: boolean) => {
+  let hasEntireCheckoutDiscount = false;
+  let hasOncePerOrderVoucher = false;
+  let discountedLinesCount = 0;
+
+  for (const line of taxBase.lines) {
+    const isLineDiscounted = checkIfIsDiscountedLine(isDiscounted, line);
+
+    if (isLineDiscounted) {
+      discountedLinesCount++;
+    }
+  }
```

Comment:
issue: if `isDiscounted` is `false`, then `isLineDiscounted` will always be `false` and we don't need to run this `for...of` loop, and we don't have to pass `isDiscounted` to `checkifIsDiscountedLine` function.

suggestion: I would also rename `isDiscounted` to be more clear where is this value coming from, e.g. `isEntireSourceDiscounted` or something like that

```suggestion
  if (isEntireSourceDiscounted) {
    for (const line of taxBase.lines) {
    	const isLineDiscounted = checkIfIsDiscountedLine(line);

    	if (isLineDiscounted) {
	      discountedLinesCount++;
	    }
	  }
  }
```